File: back-end-bus-ticket-service/user-management-service/responses/ApiResponse.cs
namespace user_management_service.responses {
    public class ApiResponse<T>
    {
        public bool Success { get; set; }
        public string Message { get; set; }
        public T Data { get; set; }
        public string Error { get; set; }
    
        public ApiResponse(bool success, string message, T data, string error)
        {
            Success = success;
            Message = message;
            Data = data;
            Error = error;
        }
    }
}

File: back-end-bus-ticket-service/staff-dispatch-service/response/ApiResponse.py
from fastapi.responses import JSONResponse

class ApiResponse:
    def __init__(self, success: bool, message: str, data: dict = None, error: str = None):
        self.success = success
        self.message = message
        self.data = data
        self.error = error

    def to_response(self, status_code: int = 200):
        return JSONResponse(
            status_code=status_code,
            content={
                "success": self.success,
                "message": self.message,
                "data": self.data,
                "error": self.error,
            }
        )